Keyboard shortcut to toggle block level HTML editing #9723

dmahendrakar opened the pull request:
## Description
Adds keyboard short cut for block level HTML editing - Fixes #9723

## How has this been tested?
Added snapshot test.

## Screenshots <!-- if applicable -->
<img width="681" alt="Screen Shot 2019-12-15 at 2 44 08 AM" src="https://user-images.githubusercontent.com/7881389/70861592-ed401c80-1ee4-11ea-9721-8b56da6db228.png">
<img width="672" alt="Screen Shot 2019-12-15 at 2 44 56 AM" src="https://user-images.githubusercontent.com/7881389/70861708-d26ea780-1ee6-11ea-962f-4ded556852df.png">
<img width="501" alt="Screen Shot 2019-12-15 at 3 27 48 AM" src="https://user-images.githubusercontent.com/7881389/70862031-90943000-1eeb-11ea-89f9-b4af1abdc134.png">

## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

Commit:
Keyboard shortcut to toggle block level HTML editing #9723

Commit:
Keyboard shortcut to toggle block level HTML editing #9723

dpkm95 commented:
Working atop @MarkDavidTurner's stale pr https://github.com/WordPress/gutenberg/pull/11941.
Few changes from the above pr include-
- used onToggleBlockMode instead of onToggleEditor to be consistent with WordPress function naming.
- used access('m') instead on primaryAlt('v') as it's closest to global block mode toggle secondary( 'm' ). Alternate option is ctrl('m') @talldan is that a better option?

TODO: Handle reusable block case. @talldan can you point me to reference how to find out if block is a reusable type?

Commit:
Ignore toggle for reusable block

Commit:
Ignore toggle for reusable block

Commit:
Renaming the help text to match global block toggle text.

Commit:
Renaming the help text to match global block toggle text.

dpkm95 commented:
I found one issue when manually testing the feature. It breaks for case of group of blocks - when I try to edit the html and toggle block mode I get *block-invalid-warning*
<img width="830" alt="Screen Shot 2019-12-15 at 10 39 52 AM" src="https://user-images.githubusercontent.com/7881389/70867169-3fa22d00-1f27-11ea-8245-6e7278e609b4.png">

I don't get why the BlockCompare shows difference that I didn't introduce - I just changed text from 'para1 changed' to 'para1'. Can someone explain or point me how to debug this?

Commit:
Trying move to trash as a link (#19138)

ockham committed:
Base Styles: Import colors into variables (#19159)

`@import "./colors";` from within `_variables.scss`, since the latter [depends](https://github.com/WordPress/gutenberg/blob/3687e8b992b6462044eead19944cf0c6ea41a1e3/packages/base-styles/_variables.scss#L36-L40) on `$dark-gray-900` which is [defined](https://github.com/WordPress/gutenberg/blob/bfd1117846c7802d62dbba046c27968d961781e7/packages/base-styles/_colors.scss#L7) in the former.

jasmussen committed:
Try: Default cursor for Select Tool (#19157)

The select tool doesn't let you select text. It's only until you click twice on the same block, press Enter, or select the Edit tool, that you edit text again. Therefore the text-beam cursor is inaccurate.

This PR changes that cursor to the default arrow cursor.

youknowriad committed:
Refactor the MediaReplaceFlow component to use Dropdown (#19126)

jasmussen committed:
Image Block: Revert circle mask, add "rounded" (#19028)

* Image Block: Revert circle mask, add "rounded"

Fixes #16130, notably a followup to the issue outlined in https://github.com/WordPress/gutenberg/issues/16130#issuecomment-555451523.

The "circle mask" variation has the benefit of creating a perfect circle out of any shape image. However it doesn't change the dimensions of the image, so the tradeoff is that the image footprint is still large, and resize handles now float in air.

This PR changes it to a "rounded" style, which sets a very high border radius ensuring that the smallest sides on a rectangular image are perfectly rounded, creating a pill-shape. When the image is perfectly square, the variation will be perfectly circular. This ensures that the variation is useful both when images are rectangular, it ensures the resize handles are appropriately placed, it fixes an issue where Microsoft Edge did not render the mask properly, and it indicates to the user that if they want a perfect circle, they can upload a square image, or change the dimensions in the sidebar.

* Add back-compat.

etoledom committed:
Conditionally hide Image Size option when Image object is not present (#19037)

This avoids showing the option when it won't do anything.

aduth committed:
Block Editor: Test ContrastChecker notices by string comparison (#19169)

jorgefilipecosta committed:
Add end 2 end tests InnerBlocks renderAppender (#14996)

* Add end 2 end tests InnerBlocks renderAppender

* Fixed intermittent problem.

* Address reviews

aduth committed:
E2E Test Utils: Remove unused KeyboardMode file (#19166)

aduth committed:
Compat: Add Trac ticket link to render_callback shim

mmtr committed:
Gallery block: Add images size selector (#18581)

Adds an image size selector to the gallery block that allows users to choose the size of the images. It defaults to the large size.

ZebulanStanphill committed:
Audio block: don't render empty audio tag. (#18850)

Commit:
Expanding on sentence case to everywhere else (#18758)

* Expanding on sentance case to everywhere else

So far we have only done this in the tooltips and in menus. If sentance case of right approach we should consider taking this across all areas.

Included in this:

- Side panel
- Block library
- Settings

There may be areas missed, but gives a starting view to see if everyone agrees to go forward with this across everything. If we do, then we need a patch for core.

Expands on #16764

* Catch a few missed

* Update tag

* Test catches

Fixes test issues.

* Catches the test fails for sentance case.

* Test fix

* Update index.js

* Catch some more test fails

* Test fixes

* Override WP labels for post and page types

… to conform to sentence-style capitalisation across the editor UI.

(This is a quick commit to show the idea. Do expand on it and polish.)

From Slack: "If we used this it would have to be properly marked [e.g.
@TODO] and commented so that before the next core release we could
revisit these labels as a whole."

* gutenberg_override_posttype_labels: small fixes

epiqueras committed:
Site Title: Remove save button. (#19155)

epiqueras committed:
Edit Site: Add package with barebones site editor screen. (#19054)

* Edit Site: Bootstrap package.

* Edit Site: Link package.

* Edit Site: Load assets in new Edit Site Page.

* Edit Site: Render basic editor.

talldan committed:
Add nav block e2e tests (#19189)

* Add tests for adding navigation blocks using two different methods

* Avoid showing placeholder until pages are loaded

* Fixes for tests

* Revert "Avoid showing placeholder until pages are loaded"

This reverts commit 0dd1a3669f43e72f11a84e1245d7e6628c19d517.

* Fix various issues with tests

* Handle console warning

* Use a mocked response from the pages endpoint

* Also mock the search response

* Try fixing e2e tests when run on travis CI.

* Update button label

Copons committed:
Update the copy of the Experiments page (#18233)

* Add an additional warning for the FSE experiment setting

* Updated warning text

* Update the copy some more

gziolo committed:
Block Editor: Improve the way terms are matched in the block inserter (#19122)

* Block Editor: Get categories from the store directly in the inserter

* Block Editor: Extract searchItems into its own file

* Improve the way terms are matched in the block inserter

gziolo committed:
Fix the unit tests which diverged after merging PR not rebased with master

cameronvoell committed:
activeFormats added to fix mobile paste crash (#19190)

ellatrix committed:
Verse block: fix white space (#19173)

ellatrix committed:
RichText: fix isComposing for Safari (#19171)

* RichText: fix isComposing for Safari

* Add comment

epiqueras committed:
Env: Support running arbitrary commands and use it for linting and server registered fixtures. (#18986)

* Env: Support running arbitrary commands and use it for linting and server registered fixtures.

* Travis: Provide NodeGit with correct C bindings.

* Env: Fix tests.

* Travis: Add clarifying comments to C lib includes.

ellatrix committed:
RichText: rewrite withFilters with hooks (#19117)

* wip

* useDispatchWithMap

* wip

* Clean up

* Clean up

* Clean up

* works

* Clean up

* Clean up

* Remove extra selectors

* Remove hooks dependency

* Rename experimentalProps

* Mark useDispatchWithMap unstable

* Don't pass dispatchProps

* Optimise

* Add note

nukaga committed:
Update columns index.js adding translatable (i18n) parts (#18162)

* Update index.js

Adding translatable (i18n) parts

* add translators comments

ellatrix committed:
Native: object fallback for element.style (#19196)

mchowning committed:
Avoid setting alignSelf for image placeholders (#18793)

This was causing the placeholders to not display

Commit:
RangeControl: Fix initialPosition to accept 0 (#18611)

* RangeControl: Fix initialPosition to accept 0

This update fixes the `initialPosition` prop for `RangeControl` to properly handle a value of `0`. Previously, the fallback value was checking for truthiness, which does not allow for `0` to work.

Being able to set the value to `0` is important with something like a range slider, as you may not have
an initial value, but you would like the slider to start from the very beginning (with an example range of 0-20).

* Refactor initialPosition valoue check. Rename initialValue story

Commit:
Adjustments for welcome guide (#19195)

* Adjustments for welcome guide

- Reduces header to have text on desktop on one line.
- Reduces line-height on header and makes it number over px.
- Tried to avoid widows but needs longer lasting solution.

* Adjust to break-small

Props @jasmussen for fix

jorgefilipecosta committed:
Fix: RangeControl Snapshots (#19202)

mchowning committed:
Pass formatTypes to FormatEdit component (#19200)

aduth committed:
Data: Remove unused forceRender argument (#19206)

jkmassel committed:
Add Inserter testID (#18832)

* Add Inserter testID

* Add inline documentation for testID

draganescu committed:
Link images in media + text (#18139)

* adds the same link functionality as the image block has to the media text block

* fixes open in new tab

* refactored ImageURLInputUI as a separate block editor component for use in the image and media+text blocks

* encapsulates all ImageURLInputUI related functionality into the component itself

* removes the ImageURLInputUI functionality from Image

* removes useless exports from ImageURLInputUI

* readme updates

* corrects the export for ImageURLInputUI

* removes useless onSetNewTab function from Image

* removes useless exports from MediaTextEdit

* removes default value for linkDestination in MediaText

* replaces Toolbar with ToolbarGroup as wrappwe for ImageURLInputUI

* destructures some props and removes Toolbar dep

* makes code more readable in media+text save function

* make image a function on save so we render only if media is image

Co-Authored-By: Jorge Costa <jorge.costa@automattic.com>

* attachment or media file link updates on file change

Commit:
[RNMobile] Stepper component (#18864)

* Stepper Control

* Stepper component

* Update docs

* RNMobile - Switch RangeControl with StepperControl in Spacer Block

* RNMobile - StepperControl accessibility changes and README file

* [RNMobile] - Restore RangeControl in the Spacer block

mmtr committed:
Font Size Picker: Add default size (#18273)

Adds a default option to the font size picker component that unsets any font size.

Previously, the font size selector was falling back to the normal size, but this size might have not been defined by theme.

scruffian committed:
Media Replace Flow: Don't show the URL option unless there is a handler (#19063)

ellatrix committed:
Accessibility: add role=group to block wrapper element (#19213)

ellatrix committed:
Accessibility: don't tab to block drag handle (#19211)

* Accessibility: don't tab to block drag handle

* Adjust e2e tests

aduth committed:
Edit Post: Fix Welcome Guide modal display for Internet Explorer (#19201)

* Components: Remove guide contents centering

Doesn't appear to be needed, and causes overflow issues in IE due to Flexbox bug

See: https://github.com/philipwalton/flexbugs#flexbug-2

* Components: Use explicit display for guide finish button

* Components: Ensure modal is styled as positioned element

* Edit Post: Vertically center welcome guide image

epiqueras committed:
Env: Fix Vim typo? (#19218)

`1cq` => `1`

The value only needs to be non-empty so nothing should change here.

dpkm95 commented:
@talldan can you please help me with direction to proceed here, I think this is either a bug in existing conversion logic or I have used it wrongly.

jorgefilipecosta committed:
Update: Round position attributes on cover focal point save (#19183)

glendaviesnz committed:
Fix/title encoding bug (#19187)

* Switch to decodeEntities instead of unescape
* Modify test case for decode util to account for leading zeros on encoded entities

dpkm95 commented:
I got input from @youknowriad that the shortcut handler should check whether the html mode is supported for the block before switching.

What is right user experience in case block doesn't support html?
- Hide the Edit as HTML shortcut in components menu?
- Show the Edit as HTML  shortcut but disable it.
- Show the Edit as HTML shortcut have the user click it and show notice in to indicate block doesn’t support html.

mkaz committed:
Move ESNext to default code example shown (#17873)

* Clarify wording around ES5/ESNext usage

* Update wording around ES5 and ESNext usage

* Move ESNext to default code

- Also remove 05 numbering of examples, does not exist in examples repo
And 05 in the examples repo is the recipe card.

* Move ESNext as default example

Update loading of dependencies and version to match examples repo,
using the asset_file from build script.

* Fix typo

* Apply suggestions from code review

Switch JS to JSX for code syntax

* Fix plugin doc block

* Update docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md

Co-Authored-By: Andrés <nosolosw@users.noreply.github.com>

* Update docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md

Co-Authored-By: Andrés <nosolosw@users.noreply.github.com>

talldan commented on packages/edit-post/src/components/keyboard-shortcut-help-modal/config.js:
```diff
@@ -124,6 +124,10 @@ const blockShortcuts = {
 		},
+		{
+			keyCombination: access( 'm' ),
+			description: __( 'Switch between Visual Editor and Code Editor.' ),
```
It'd be good to differentiate this from the other shortcut a bit more. How about something like the following:
```suggestion
			description: __( 'Switch between editing the block visually and editing its HTML.' ),
```

dpkm95 replied:
@talldan is ctrl('m') a better option than access( 'm' )?

youknowriad commented:
> What is right user experience in case block doesn't support html?

I think it's already hidden in the menu if the block doesn't support it.

talldan commented:
> I got input from @youknowriad that the shortcut handler should check whether the html mode is supported for the block before switching.

@dpkm95 Thanks for taking on another issue! It looks like you have some checks in your PR already:
https://github.com/WordPress/gutenberg/pull/19152/files#diff-ad3bcfe81a33d76e53a33739c9c89b45R171

but they need to match the logic used to hide the menu item when HTML is not supported:
https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-settings-menu/block-mode-toggle.js#L16

For particular blocks like the Columns block, the `Edit as HTML` option isn't supported, so your code needs to use `hasBlockSupport` to check for that:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/columns/index.js#L26

It looks like it's possible to disabled HTML editing across the entire editor as well, so that should also be checked:
https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-settings-menu/block-mode-toggle.js#L39

I don't think you need to worry about changing the menu, it already hides the option when needed.

jasmussen committed:
Make Placeholder component "element-query-like" responsive. (#18745)

Props @jameslnewell.

noisysocks committed:
Deprecate '@wordpress/nux' package (#18981)

nosolosw committed:
Add a pre-commit hook to check whether API docs are updated (#18820)

gziolo committed:
Project management: Add prepublish packages command for npm releases (#19214)

* Project management: Add prepublish packages command for npm releases

* Adjust the message displayed when the process ends

* Update bin/commander.js

Co-Authored-By: Riad Benguella <benguella@gmail.com>

* Improve the clean temporary directories step

ellatrix committed:
Writing flow: improved tabbing for Edit mode (#19235)

Copons committed:
Navigation: Add font size attribute (#19127)

Adds the font size style attributes (both named and custom) to the Navigation block.

gziolo committed:
Documentation: Do not expose documentation for mobily only components (#19236)

gziolo committed:
Update changelogs

gziolo committed:
chore(release): publish

 - @wordpress/annotations@1.9.0
 - @wordpress/api-fetch@3.8.0
 - @wordpress/babel-preset-default@4.8.0
 - @wordpress/base-styles@1.1.0
 - @wordpress/block-directory@1.2.0
 - @wordpress/block-editor@3.4.0
 - @wordpress/block-library@2.11.0
 - @wordpress/blocks@6.9.0
 - @wordpress/components@8.5.0
 - @wordpress/compose@3.9.0
 - @wordpress/core-data@2.9.0
 - @wordpress/data-controls@1.5.0
 - @wordpress/data@4.11.0
 - @wordpress/date@3.7.0
 - @wordpress/docgen@1.6.0
 - @wordpress/e2e-test-utils@4.0.0
 - @wordpress/e2e-tests@1.9.0
 - @wordpress/edit-post@3.10.0
 - @wordpress/edit-widgets@0.9.0
 - @wordpress/editor@9.9.0
 - @wordpress/element@2.10.0
 - @wordpress/env@0.2.0
 - @wordpress/eslint-plugin@3.3.0
 - @wordpress/format-library@1.11.0
 - @wordpress/jest-preset-default@5.3.0
 - @wordpress/jest-puppeteer-axe@1.5.0
 - @wordpress/list-reusable-blocks@1.10.0
 - @wordpress/media-utils@1.4.0
 - @wordpress/notices@1.10.0
 - @wordpress/nux@3.9.0
 - @wordpress/plugins@2.9.0
 - @wordpress/rich-text@3.9.0
 - @wordpress/scripts@6.1.0
 - @wordpress/server-side-render@1.5.0
 - @wordpress/token-list@1.8.0
 - @wordpress/url@2.9.0
 - @wordpress/viewport@2.10.0

youknowriad committed:
Merge the Button and IconButton into a single component (#19193)

gziolo committed:
Publishing: Remove obsolete build step for Lerna updated check

aduth committed:
Block Editor: Consistently render IgnoreNestedEvents for block appender (#19135)

* Block Editor: Consistently render IgnoreNestedEvents for block appender

* Block Editor: Fix grammar in BlockListAppender comment

* Block Editor: Capture button click focus in BlockListAppender

Firefox, Safari compatibility

* Block Editor: Remove redundant IgnoreNestedEvents from InnerBlocks appender

Handled now via ancestor BlockListAppender in which this component would be rendered

MarcoZehe committed:
Accessibility: Make text alignment items radio menu items. (#19233)

* Accessibility: Make text alignment items radio menu items.

Pass a role from the alignment toolbar to the dropdown menu. Make the dropdown menu aware of this role on each control and set it if present. Also set aria-checked on the active control if the role calls for it. Fixes #18721.

* Fix copy&paste error.

* Fix test.

* Also fix the block alignment toolbar.

* Also include block vertical alignment controls.

* Make the roles conditional on them being in a dropdown

Commit:
Removes reusable block option from navigation link (#19250)

Fixes #18318 by removing reusable blocks.

mkaz committed:
Docs: Add Developer Tools setup in Getting Started (#19074)

* Add Editor setup in Getting Started

* Edits via @griffbad review

* Update wording per @jsnajdr review

* Update getting-started with ESLint and TypeScript info

Commit:
Updating help text

Commit:
Updating help text

Commit:
Added block html support check

Commit:
Added block html support check

dpkm95 commented:
@talldan @youknowriad I've added the checks, now it works well.
Waiting for https://github.com/WordPress/gutenberg/pull/19100 to be merged.

talldan commented:
> Meanwhile, is ctrl('m') a better option than access( 'm' )?

@dpkm95 I think that shortcut is already used by some browsers—Firefox I think uses it for muting/umuting audio, and Edge uses it to display the reading list.

talldan reviewed (approved):
Nice work @dpkm95! Congrats on your first contribution to the project 🎉

talldan commented:
Oh, just one thing that might be useful, if you write `Fixes #9723` or `Closes #9723` in the Pull Request description, #9723 will be closed automatically as resolved when this PR is merged.

noisysocks committed:
Add more documentation for @wordpress/env (#19194)

* @wordpress/env: Add more documentation

* Typo: Port should be 8888

Co-Authored-By: Enrique Piqueras <epiqueras@users.noreply.github.com>

* Typo: s/version/versions/

Co-Authored-By: Enrique Piqueras <epiqueras@users.noreply.github.com>

* Typo: Port should be 8888

Co-Authored-By: Enrique Piqueras <epiqueras@users.noreply.github.com>

* Typo: s/wordpress/tests-wordpress/

Co-Authored-By: Enrique Piqueras <epiqueras@users.noreply.github.com>

Co-authored-by: Enrique Piqueras <epiqueras@users.noreply.github.com>

jorgefilipecosta committed:
Add tabs for gradient and color (#19133)

jbinda committed:
[RNMobile] fix group appender behaviour (#18564)

* fix show appender in group block

* add isOpen check to conditionally show/hide appender button under group block

* render Separator in Inserter

* pass showSeparator

* code-styling fix

* remove unused file and fix insertion point

* nit: remove unnecessary break line

Co-authored-by: Drapich Piotr <drapich.piotr@gmail.com>
Co-authored-by: Luke Walczak <lukasz.walczak.pwr@gmail.com>

ellatrix committed:
Popover: reposition on click (#19268)

* Popover: reposition on click

* Leave comment

talldan committed:
Remove unused `blocks-font-size` classname (#19208)

glendaviesnz committed:
Make validation of block html tags and attributes case insensitive (#19207)

* Make validation of block html tags and attributes case insensitive
* Optimize case-insensitive checks with a for loop and deferring checks for minority case
* Add test cases for isEqualTagAttributePairs and isEqualTokensOfType checking the case-insensitive behavior
* Blocks: Validation: Avoid creation of actualAttributes object map

Co-authored-by: @glendaviesnz, @mmtr and @aduth

dhurlburtusa committed:
Change back to Prev Destination but with Correct URL (+1 squashed commit) (#18423)

Squashed commits:
[4fd170c885] Fix Broken Link; Misc Grammar Fixes; Indentation Fix

It looks like the link was trying to point to https://developer.wordpress.org/block-editor/tutorials/javascript/loading-javascript/. But I think is would be better to point to https://developer.wordpress.org/block-editor/tutorials/javascript/extending-the-block-editor/ which has the correct "Step 2" which also includes the dependencies array for the code.

akkspros committed:
Coding Standards: Fixed (#19272)

aduth committed:
Block Editor: Remove block inserter shortcuts (#19045)

MarcoZehe committed:
Accessibility: Improve Welcome guide and modal component. (#19261)

ellatrix committed:
Popover: remove buffer options (#19283)

dpkm95 commented on packages/block-editor/src/components/block-actions/index.js:
```diff
@@ -165,6 +167,14 @@ export default compose( [
+				const isCodeEditingEnabled = select( 'core/editor' ).getEditorSettings().codeEditingEnabled;
+
+				if ( hasBlockSupport( block.name, 'html', true ) && isCodeEditingEnabled && ! isLocked && ! isReusableBlock( block ) ) {
+					toggleBlockMode( clientIds );
```
@talldan should I use clientIds[0] to be consistent `const block = blocks[ 0 ];`?

pento committed:
Block API: Pasting content that results in a new block shouldn't be treated as inline content. (#19084)

* Only treat the paste as inline if it results in a paragraph block.

* Whitelist blocks that can be treated as inline pastes.

* Add a supports.pasteTextInline property to blocks, and some tests.

* Mark pasteTextInline unstable

Co-authored-by: Ella van Durpe <ella@automattic.com>

ellatrix committed:
Navigation mode: enable on tab if no block is selected (#19238)

* Navigation mode: enable on tab if no block is selected

* Ensure edit mode with enter from title

* Let inserting block always trigger Edit mode

MarcoZehe committed:
Guide Accessibility: Change the aria-current expression to be a step instead of a boolean (#19290)

* Guide Accessibility: Change the aria-current expression to be a step instead of a boolean

* Add inline documentation for the aria-current attribute.

StevenDufresne committed:
Block Directory: Use the block's title for alt text on block directory plugin items. (#19263)

Merged pull request #19263 into master.

* Block Directory: Use the block's title to construct an alt text for the block's icon when they are using an image file.

Fixes:
https://github.com/WordPress/gutenberg/pull/16524/files#r324268820

* Add an accompanying translator comment to explain the placeholder.

youknowriad committed:
Add a new keyboard shortcut package (#19100)

youknowriad committed:
Remove components-icon-button classname (#19241)

ellatrix committed:
Navigation mode: polish enabling when no block is selected (#19298)

* Navigation mode: polish enabling when no block is selected

* Account for title

* Add e2e tests

ellatrix committed:
Navigation mode: fix tabbing to title (#19305)

* Navigation mode: fix tabbing to title

* Remove unused ref

mkaz committed:
Fix link to loading javascript (#19280)

ellatrix committed:
Lighter block DOM: contextual toolbar in popover (#18779)

* Lighter block DOM: contextual toolbar in popover

* Movers in toolbar (same style as master)

* Smooth repositioning on move

* Fix e2e test

* Only translate for animation

* Address feedback

* Specify subpixel direction to work around Chrome bug

* Remove dead CSS

* Polish alignment position

* Fix toolbar position for RTL

jorgefilipecosta committed:
Fix: Only show available image sizes (#19301)

epiqueras committed:
Jest Preset Default: Update preset file extension for inclusion in NPM deployments. (#19306)

ellatrix committed:
Popover: don't render fallback anchor if anchorRef is defined (#19308)

* Popover: don't render fallback anchor if anchorRef is defined

* include anchorRect

ellatrix committed:
Writing flow: simplify & fix tabbing out of block (#19312)

* Writing flow: simplify & fix tabbing out of block

* Fix Storybook

abdel-h committed:
Fixed dates alignments in the picker (#19294)

By removing a negative margin, that probably was used to fix an issue but it's no longer needed after another recent change was made
I also added a basic storybook for the component.

marecar3 committed:
[RNMobile] Fix Button and useShortcut for mobile (#19315)

youknowriad commented:
now, that the keyboard shortcuts refactor landed, this probably needs a rebase.

abdel-h committed:
Update CustomSelectControl - using items width instead of 100% #19130 (#19150)

Commit:
Removes title case from alignments for text and image (#18757)

youknowriad committed:
Deprecate IconButton and replace its usage with Button (#19299)

ellatrix committed:
RichText: split out boundary style calculation (#19319)

* RichText: split out boundary style calculation

* Add some tests

ellatrix committed:
Writing flow: add some docs (#19314)

marecar3 committed:
[RNMobile] - Fix failed tests on Gutenberg mobile repo (#19323)

youknowriad committed:
Refactor the visual editor shortcuts to use the keyboard-shortcuts package (#19318)

youknowriad committed:
Refactor edit post shortcuts to use the keyboard-shortcuts package (#19320)

youknowriad committed:
Refactor KeyboardShortcuts component to rely on useKeyboardShortcut hook (#19325)

youknowriad committed:
Fix post title style

marecar3 committed:
[RNMobile] Bring master back to normal (#19328)

dpkm95 commented:
@youknowriad sure, I'm working on it.
While going through the keyboard shortcuts refactor I realized we need to handle case where current block supports html but it's child doesn't. Do you see a possibility for such a case? Is it already handled by toggleBlockMode?

Commit:
Rebasing with 19100

Commit:
Rebasing with 19100

dpkm95 commented:
Opened another pr for the same: https://github.com/WordPress/gutenberg/pull/19329/
